File: integrations/resolve-workflow/bridge_client.py
import json
import urllib.request

BRIDGE_BASE = "http://127.0.0.1:8544"


def fetch_json(path: str):
    req = urllib.request.Request(f"{BRIDGE_BASE}{path}", method="GET")
    with urllib.request.urlopen(req, timeout=3) as res:
        return json.loads(res.read().decode("utf-8"))


def list_projects():
    return fetch_json("/api/nle/projects")


def list_clips(project_id: str | None = None):
    query = f"?projectId={project_id}" if project_id else ""
    return fetch_json(f"/api/nle/clips{query}")

File: integrations/avid-mediacentral-panel/bridge-client.js
const BRIDGE_BASE = "http://127.0.0.1:8544";

export async function getBridgeHealth() {
  const res = await fetch(`${BRIDGE_BASE}/api/nle/health`);
  if (!res.ok) {
    throw new Error(`Bridge unavailable: ${res.status}`);
  }
  return res.json();
}

export async function getProjects() {
  const res = await fetch(`${BRIDGE_BASE}/api/nle/projects`);
  if (!res.ok) {
    throw new Error(`Unable to fetch projects: ${res.status}`);
  }
  return res.json();
}

export async function getClips(projectId) {
  const query = projectId ? `?projectId=${encodeURIComponent(projectId)}` : "";
  const res = await fetch(`${BRIDGE_BASE}/api/nle/clips${query}`);
  if (!res.ok) {
    throw new Error(`Unable to fetch clips: ${res.status}`);
  }
  return res.json();
}
